src/foam/nanos/analytics/mixpanel/MixpanelAnalyticEventAction.js

```diff
@@ -91,6 +98,17 @@ foam.CLASS({
           }
         }, "Send message to mixpanel");
       `
+    },
+    {
+      name: 'isWhitelisted',
+      args: 'X x, AnalyticEvent event',
+      javaType: 'Boolean',
+      javaCode: `
+        ServiceProvider spid = (ServiceProvider) ((DAO) x.get("serviceProviderDAO")).find((String) x.get("spid"));
+        String[] whitelist = spid.getMixpanelWhitelist();
+        var pred = IN(foam.nanos.analytics.AnalyticEvent.NAME, whitelist);
```

Comment:
I'm not super-keen on the performance of this implementation since it is going to do a linear scan for each analytic event.